pipenv/utils/resolver.py

```diff
@@ -980,19 +979,19 @@ def venv_resolve_deps(
     target_file.close()
     cmd.extend(["--write", make_posix(target_file.name)])
     with temp_environ():
-        os.environ.update({fs_str(k): fs_str(val) for k, val in os.environ.items()})
+        os.environ.update({k: val for k, val in os.environ.items()})
```

Comment:
This statement seems to have no effect after the refactor -- we should consider if val needs to be cast to a `str` or if this line can be removed entirely.

Reply:
Good catch!
```
In [4]: os.environ.update({"foo": Foo()})
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-b63c173a9c62> in <module>
----> 1 os.environ.update({"foo": Foo()})

/usr/lib/python3.9/_collections_abc.py in update(self, other, **kwds)
    939         if isinstance(other, Mapping):
    940             for key in other:
--> 941                 self[key] = other[key]
    942         elif hasattr(other, "keys"):
    943             for key in other.keys():

/usr/lib/python3.9/os.py in __setitem__(self, key, value)
    682     def __setitem__(self, key, value):
    683         key = self.encodekey(key)
--> 684         value = self.encodevalue(value)
    685         putenv(key, value)
    686         self._data[key] = value

/usr/lib/python3.9/os.py in encode(value)
    754         def encode(value):
    755             if not isinstance(value, str):
--> 756                 raise TypeError("str expected, not %s" % type(value).__name__)
    757             return value.encode(encoding, 'surrogateescape')
    758         def decode(value):

TypeError: str expected, not Foo

```
This works:
```
os.environ.update({"foo": str(Foo())})
```